aiokafka/group_coordinator.py

```diff
@@ -0,0 +1,847 @@
+                using Kafka's group managementment facilities. Default: 30000
+            retry_backoff_ms (int): Milliseconds to backoff when retrying on
+                errors. Default: 100.
+        """
+        self._client = client
+        self._session_timeout_ms = session_timeout_ms
+        self._heartbeat_interval_ms = heartbeat_interval_ms
+        self._retry_backoff_ms = retry_backoff_ms
+        self.generation = OffsetCommitRequest.DEFAULT_GENERATION_ID
+        self.member_id = JoinGroupRequest.UNKNOWN_MEMBER_ID
+        self.group_id = group_id
+        self.coordinator_id = None
+        self.rejoin_needed = True
+        self.needs_join_prepare = True
+        self.loop = loop
+        # rejoin group can be called in parallel
+        # (from consumer and from heartbeat task), so we need lock
+        self._rejoin_lock = asyncio.Lock(loop=loop)
+        self._enable_auto_commit = enable_auto_commit
+        self._auto_commit_interval_ms = auto_commit_interval_ms
+        self._assignors = assignors
+        self._subscription = subscription
+        self._partitions_per_topic = {}
+        self._cluster = client.cluster
+        self._cluster.request_update()
+        self._cluster.add_listener(self._handle_metadata_update)
+        self._auto_commit_task = None
+        self._closing = asyncio.Future(loop=loop)
+
+        self.heartbeat_task = ensure_future(
+            self._heartbeat_task_routine(), loop=loop)
+
+        if self._enable_auto_commit:
+            interval = self._auto_commit_interval_ms / 1000.0
+            self._auto_commit_task = ensure_future(
+                self.auto_commit_routine(interval), loop=loop)
+
+    @asyncio.coroutine
+    def close(self):
+        """Close the coordinator, leave the current group
+        and reset local generation/memberId."""
+        self._closing.set_result(None)
+        if self._auto_commit_task:
+            try:
+                yield from self._auto_commit_task
+            except asyncio.CancelledError:
+                pass
+            self._auto_commit_task = None
+
+        if self.heartbeat_task:
+            self.heartbeat_task.cancel()
+            try:
+                yield from self.heartbeat_task
+            except asyncio.CancelledError:
+                pass
+            self.heartbeat_task = None
+
+        if not (yield from self.coordinator_unknown()) and self.generation > 0:
+            # this is a minimal effort attempt to leave the group. we do not
+            # attempt any resending if the request fails or times out.
+            request = LeaveGroupRequest(self.group_id, self.member_id)
+            try:
+                yield from self._send_req(self.coordinator_id, request)
+            except Errors.KafkaError as err:
+                log.error("LeaveGroup request failed: %s", err)
+            else:
+                log.info("LeaveGroup request succeeded")
+
+        self.generation = OffsetCommitRequest.DEFAULT_GENERATION_ID
```

Comment:
Can we reuse the coordinator after closure? I think it's better to create new one and just forget about those params.